### PR TITLE
Add species to admin/issue 338

### LINKED ISSUE #339: Species: edit current species item.
On the species list page, click 'edit' icon to edit the detail of a species. 

https://14bvv4.axshare.com/#g=1&p=species&c=1
Screen Shot 2020-08-04 at 6 19 32 AM

### PR DESCRIPTION
**Pull request by elfiyang16**
resolves #339 
Aug-18-2020 21-04-26

### CODE REVIEW

File: client/src/api/treeTrackerApi.js [hunk 1]
```diff
@@ -82,149 +75,176 @@ export default {
...
         active: 0,
         valueFactor: 0,
-      })
+      }),
+    })
+      .then(handleResponse)
+      .catch(handleError)
+  },
+  /* edit specie */
+  editSpecie(id, name, desc) {
```
Comment (nmcharlton) [inline client/src/api/treeTrackerApi.js]: Should be `editSpecies` (singular of species is species)

File: client/src/components/SpecieTable.js [hunk 2]
```diff
@@ -88,17 +94,21 @@ const styles = (theme) => ({
...
-  React.useEffect(() => {
-    props.speciesDispatch.loadSpeciesList()
-    console.log(props.speciesState.speciesList)
-  }, [])
   const { classes } = props
 
   const [page, setPage] = React.useState(0)
   const [rowsPerPage, setRowsPerPage] = React.useState(5)
+  const [isEdit, setIsEdit] = React.useState(false)
+  const [specieEdit, setSpecieEdit] = React.useState(undefined)
```
Comment (nmcharlton) [inline client/src/components/SpecieTable.js]: "species", not "specie"

File: client/src/components/SpecieTable.js [hunk 3]
```diff
@@ -108,6 +118,11 @@ const SpecieTable = (props) => {
     setPage(0)
   }
 
+  const handleEdit = (specie) => {
```
Comment (nmcharlton) [inline client/src/components/SpecieTable.js]: species

File: client/src/components/SpecieTable.js [hunk 4]
```diff
@@ -205,10 +220,102 @@ const SpecieTable = (props) => {
...
+    setIsEdit(false)
+    await editSpecie({ id: specieEdit.id, name: specieEdit.name, desc: specieEdit.desc })
+    loadSpeciesList()
+    setFinishEdit(false)
+    setSpecieEdit(undefined)
+  }
+
+  return (
+    <Dialog open={isEdit} aria-labelledby="form-dialog-title">
+      <DialogTitle id="form-dialog-title">Specie Detail</DialogTitle>
```
Comment (nmcharlton) [inline client/src/components/SpecieTable.js]: "Species Detail"

File: client/src/models/species.js [hunk 5]
```diff
@@ -61,20 +66,28 @@ const species = {
...
+            return a
+          }
+        }, undefined)
       }
     },
-	},
+    /*
+     * to edit the specie
+     */
+    async editSpecie(payload, state) {
```
Comment (nmcharlton) [inline client/src/models/species.js]: `editSpecies`

**Review by nmcharlton — CHANGES_REQUESTED** [review]
Looks great! Only comment is that the singular of _species_ is also _species_ 😄
I've only put comments against some of them.

**Comment by nmcharlton** [comment]
Also, if you write "resolves #339" in the description, the issue will be auto-closed when the PR is merged. 😃 

https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: client/src/api/treeTrackerApi.js (hunk 1 above)
Comment (elfiyang16) [reply client/src/api/treeTrackerApi.js]: lol thanks! That's embarrassing haha why does it not have a pattern 👍

**Comment by elfiyang16** [comment]
> Also, if you write "resolves #339" in the description, the issue will be auto-closed when the PR is merged. 😃
> 
> https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

that's a nice trick!

**Pull request merged**